### PR TITLE
working implementation of communication RFC

### PR DESCRIPTION
**Pull request by alfonsograziano**
### Ticket:

- [22072](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/22072)

### Description

Implemented secure communication RFC 

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [X] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/integrator-api/app.ts [hunk 1]
```diff
@@ -84,6 +84,38 @@ export async function build(opts?: FastifyServerOptions) {
...
+    const { token } = request.query as any;
+
+    if (!token) {
+      return reply.code(400).send({ error: "Token not provided" });
+    }
+
+    // We need to store this somehow in an env variable
+    const jwksRegistry = {
+      payments: "http://localhost:8001/.well-known/jwks.json",
+    };
```
Comment (peschina) [inline apps/integrator-api/app.ts:99]: We would need just the payments domain (or similar) in an env variable

File: apps/payments-api/app.ts [hunk 2]
```diff
@@ -98,5 +102,42 @@ export async function build(opts?: FastifyServerOptions) {
...
+    const { publicKey } = await readOrGenerateKeyPair("payments-api");
+    return getJWKSRoute(publicKey);
+  });
+
+  // This is a test route that calls the integrator with a callback
+  app.get("/call-integrator", async () => {
+    // const privateKey = await readLocalPrivateKey(
+    //   join(__dirname, "private.key"),
+    // );
+    const { privateKey } = await readOrGenerateKeyPair("payments-api");
```
Comment (peschina) [inline apps/payments-api/app.ts:117]: How will we actually get the private key?

**Review by peschina — APPROVED** [review]
Looks good, but we need a suitable solution for key storage and rotation

File: apps/integrator-api/app.ts (hunk 1 above)
Comment (alfonsograziano) [reply apps/integrator-api/app.ts:99]: Yess

File: apps/payments-api/app.ts (hunk 2 above)
Comment (alfonsograziano) [reply apps/payments-api/app.ts:117]: This method is meant to be used only locally (I don't want people to create multiple keypair on their machine). In the new PR to integrate it properly within payments, I'll manage that with KMS with @williamwd

**Pull request merged**